theme/_includes/head.html

```diff
@@ -4,8 +4,12 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
 
   <title>{% if page.title %}{{ page.title | escape }}{% else %}{{ site.title | escape }}{% endif %}</title>
-  <meta name="description" content="{% if page.description %}{{ page.description }}{% else %}{{ site.description }}{% endif %}">
-
+  {% for file in site.data %}
+    {% if page.dir contains file[0] %}
+    {% assign doc = file[1] %}
+      <meta name="description" content="{% if doc.description %}{{ doc.description }}{% else %}{{ site.description }}{% endif %}">
+    {% endif %}
+  {% endfor %}
```

Comment:
These dashes should help remove extra whitespace, which will result in easier-to-read HTML for people.

```suggestion
  {%- for file in site.data -%}
    {%- if page.dir contains file[0] -%}
    {%- assign doc = file[1] -%}
      <meta name="description" content="{%- if doc.description -%}{{ doc.description }}{%- else -%}{{ site.description }}{%- endif -%}">
    {%- endif -%}
  {%- endfor -%}
```